metrics/web/docs/features/firestore_emulator_for_driver_tests/01_firestore_emulator_for_driver_tests.md

```diff
@@ -0,0 +1,212 @@
+
+Now we can run the emulator with the initial data by adding the following flag `--import=import_directory`, where the `import_directory` is the directory that holds the exported data.
+
+So with that in place, we can run the emulator by using the following command:
+
+```bash
+    firebase emulators:start --only firestore --import=import_directory
+```
+
+Now the emulator is running and it is time to use it in the app's integration tests.
+
+In the `setUpAll` method of the tests we need to configure the Firestore instance to use the emulator:
+
+```dart
+setupAll(() {
+    Firestore.instance.settings(
+        host: 'localhost:8080',
+        sslEnabled: false,
+        persistenceEnabled: false,
+    );
+});
+```
+
+The specified port in the `host` argument must be equal to the emulator's port.
+
+# Design
+
+### User Interface
+
+> How users will interact with the feature (API, CLI, Graphical interface, etc.).
+
+The feature is introduced by adding additional parameters to the integration tests running command.
+
+The next table contains the definitions of the parameters with their default values.
+
+| Parameter         | Default | Description                                                           |
+| ----------------- | -------- | --------------------------------------------------------------------- |
+| **use-emulator**  | true     | Determines if the integration tests are using the `Firebase Emulator`. |
+| **emulator-port** | 8080     | Specifies the emulator's running port.                                  |
+
+The following code snippet shows an example of using the new parameters:
+
+```bash
+dart test_driver/main.dart --use-emulator --emulator-port=8081
+```
+
+Or, if you accept the defaults, you can omit them.
+
+As described in the [Prototyping](#prototyping) section, we should run integration tests once the `Firebase Emulator` started.
+
+### Database
+
+> How relevant data will be persisted.
+
+Integration tests interacts with the application that uses data from the database. 
+
+As we don't want to use the production environment, we should use the `Firebase local emulator`.
+
+The emulator has no data as a default, so we need to create necessary collections.
+
+It is not convenient to create it every time the emulator starts. So, for this purposes, we need to create all collection once and export it using the following command:
+
+```bash
+firebase emulators:export export_directory
+```
+
+_**Note:** You can place just `.` instead of the `export_directory` name and the firebase creates the directory, named `firestore_export` as a default._
+
+From now, we can run emulator with the exported data using the following command:
+
+```bash
+firebase emulators:start --import=export_directory
+```
+
+### Program
+
+> Detailed solution description to class/method level.
```

Comment:
Should we explain the way of running the `Firestore emulator` in terms of the `test_driver` here?